model.py

```diff
@@ -0,0 +1,483 @@
+        """
+        shotgrid_user_name = shotgrid_user.get("name")
+        shotgrid_user_id = shotgrid_user.get("id")
+
+        student_graduation_year = self.get_student_graduation_year(
+            shotgrid_user
+        )
+        student_year = self.get_current_student_year(student_graduation_year)
+
+        self.user_information = UserInformation(
+            shotgrid_user_name,
+            shotgrid_user_id,
+            student_year,
+            student_graduation_year,
+        )
+
+    @staticmethod
+    def get_student_graduation_year(shotgrid_user: dict) -> int:
+        """Gets the graduation year of the student. Needed for calculating which
+        year they're currently in.
+
+        Args:
+            shotgrid_user: ShotGrid user to get graduation year for
+
+        Returns:
+            int: Year in which student graduates.
+        """
+        graduation_year_string = shotgrid_user.get("sg_lichting")
+        return int(graduation_year_string[1:])
+
+    @staticmethod
+    def get_current_student_year(graduation_year: int) -> int:
+        """Gets the year the student is in. This will be 4th, 3rd or 2nd year.
+        This is necessary because each year has it's own storage server.
+
+        Args:
+            graduation_year: The year in which the student will graduate.
+
+        Returns:
+            int: The schoolyear the student is currently in.
+        """
+        current_time = datetime.datetime.now()
+        corrected_time = current_time + datetime.timedelta(days=120)
+        current_year = int(corrected_time.strftime("%Y"))
+        to_graduation_year = graduation_year - current_year
+
+        student_year = 4 - to_graduation_year
+
+        if student_year > 4:
+            student_year = 4
+
+        return student_year
+
+    def validate_project_name(self, project_name: str) -> (bool, str):
+        """Validates the project name.
+
+        Args:
+            project_name: The string name for the project.
+
+        Returns:
+            Bool: Whether or not validation is successful.
+            Str: Validation message for display in UI.
+        """
+        self.project_information.project_name = project_name
+
+        if not project_name:
+            return (
+                False,
+                "You must fill in a project name.",
+            )
+
+        legal_characters = "^[a-z_]*$"
+        contains_only_legal_characters = re.match(
+            legal_characters, project_name
+        )
+
+        if not contains_only_legal_characters:
+            return (
+                False,
+                "Project name may only use lowercase a-z and underscores.",
+            )
+
+        if project_name in self.projects:
+            return (
+                False,
+                "Project name already taken.",
+            )
+
+        return True, "Project name available!"
+
+    def validate_project_code(self, project_code: str) -> (bool, str):
```

Comment:
As mentioned in the other comment, lets split this class up into functions. But also getting rid of the tuple return.

Besides that, this is not the syntax for returning a tuple in typehints.


```python
def my_function() -> tuple[bool, str]
```

That would be the correct syntax.

if you return either a bool or a string, this would be the syntax:


```python
def my_function() -> bool | str:
```